File: LeetCode-Java/Count and Say.java
public class Solution {
    public String countAndSay(int n) {
        String result = "1";
        while(n > 1)
        {
            String newResult = "";
            int count = 1;
            char preChar = result.charAt(0);
            for(int i = 1;i <= result.length();i ++)
            {
                if(i == result.length() || result.charAt(i) != preChar)
                {
                    newResult += String.valueOf(count);
                    newResult += String.valueOf(preChar);
                    if(i != result.length())
                    {
                        count = 1;
                        preChar = result.charAt(i);
                    }
                }
                else
                    count ++;
            }
            result = newResult;
            n --;
        }
        return result;
    }
}

File: Leetcode-cpp/count_and_say.cpp
class Solution {
public:
    string countAndSay(int n) {
        string start = "1";
        for (int i = 2; i <= n; ++i) {
            start = next_string(start);
        }
        return start;
    }
    string next_string(string val) {
        string result = "";
        char current_val = val[0];
        int count = 0;
        for (int i = 0; i <= val.length(); ++i) {
            if (i == val.length()) {
                if (count > 0) result += to_string(count) + to_string((int)(current_val - '0'));
            } else if (current_val == val[i]) {
                count++;
            } else {
                if (count > 0) result += to_string(count) + to_string((int)(current_val - '0'));
                current_val = val[i];
                count = 1;
            }
        }
        return result;
    }
};